### PR TITLE
Make validation more robust on Adreno.

### PR DESCRIPTION
**Pull request by lpy**
The average of 'Textures / Fragment' should be 1 for the vulkan sample while
each 50 ms sample might vary a bit larger than 0.1 from 1.

Minor: Move some code around to avoid duplication.

Bug: b/151801290
Test: bazel run gapit -- validate_gpu_profiling --os android

### CODE REVIEW

File: gapis/trace/android/validate/validate.go [hunk 1]
```diff
@@ -117,26 +98,54 @@ func CheckEqualTo(num float64) Checker {
...
 	}
 }
 
+// CheckLargerThanZero is a checker that checks that the values are all greater than zero.
+func CheckLargerThanZero() Checker {
+	return ForeachValue(func(v float64) bool {
+		if v <= 0.0 {
+			return false
+		}
+		return true
```
Comment (pmuetschard) [inline gapis/trace/android/validate/validate.go:124]: `return v > 0.0`

File: gapis/trace/android/validate/validate.go [hunk 2]
```diff
@@ -117,26 +98,54 @@ func CheckEqualTo(num float64) Checker {
...
+	})
+}
+
+// CheckEqualTo returns a checker that checks that all returned value equal the given value.
+func CheckEqualTo(num float64) Checker {
+	return ForeachValue(func (v float64) bool {
+		if v != num {
+			return false
+		}
+		return true
```
Comment (pmuetschard) [inline gapis/trace/android/validate/validate.go:134]: `return v == num`

File: gapis/trace/android/validate/validate.go [hunk 3]
```diff
@@ -117,26 +98,54 @@ func CheckEqualTo(num float64) Checker {
...
+}
+
+// CheckAverageApproximateTo checks whether the average of the values is within
+// a margin of the given value.
+func CheckAverageApproximateTo(num, err float64) Checker {
+	return Average(func(v float64) bool {
+		if math.Abs(num - v) > err {
+			return false
+		}
+		return true
```
Comment (pmuetschard) [inline gapis/trace/android/validate/validate.go:145]: err -> margin (err should be used only for errors) and
`return mat.Abs(num - v) <= margin`

**Comment by lpy** [comment]
thanks for the feedback, ptal

**Comment by hevrard** [comment]
(just need to fix presubmit)

**Pull request merged**